.azdo/pipelines/azure-dev.yml

```diff
@@ -0,0 +1,63 @@
+# Run when commits are pushed to main
+trigger:
+  - main
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+  # setup-azd@0 needs to be manually installed in your organization
+  # if you can't install it, you can use the below bash script to install azd
+  # and remove this step
+  - task: setup-azd@1
+    displayName: Install azd
+
+  # If you can't install above task in your organization, you can comment it and uncomment below task to install azd
+  # - task: Bash@3
+  #   displayName: Install azd
+  #   inputs:
+  #     targetType: 'inline'
+  #     script: |
+  #       curl -fsSL https://aka.ms/install-azd.sh | bash
+
+  # azd delegate auth to az to use service connection with AzureCLI@2
+  - pwsh: |
+      azd config set auth.useAzCliAuth "true"
+    displayName: Configure AZD to Use AZ CLI Authentication.
+  - task: AzureCLI@2
+    displayName: Provision Infrastructure
+    inputs:
+      azureSubscription: DevOpsAzureRMConnection
+      scriptType: bash
+      scriptLocation: inlineScript
+      keepAzSessionActive: true
+      inlineScript: |
+        azd provision --no-prompt
+    env:
+      AZURE_CLIENT_ID: $(AZURE_CLIENT_ID)
+      AZURE_TENANT_ID: $(AZURE_TENANT_ID)
+      AZURE_SUBSCRIPTION_ID: $(AZURE_SUBSCRIPTION_ID)
+      SERVICE_GRAFANA_RESOURCE_EXISTS: $(SERVICE_GRAFANA_RESOURCE_EXISTS)
+      AZURE_ENV_NAME: $(AZURE_ENV_NAME)
+      AZURE_LOCATION: $(AZURE_LOCATION)
+      SERVICE_CPUADUPDATER_RESOURCE_EXISTS: $(SERVICE_CPUADUPDATER_RESOURCE_EXISTS)
+      SERVICE_ELASTIC_SEARCH_RESOURCE_EXISTS: $(SERVICE_ELASTIC_SEARCH_RESOURCE_EXISTS)
+      GITHUB_ORGANIZATION_SLUGS: $(GITHUB_ORGANIZATION_SLUGS)
+      SERVICE_UPDATEGRAFANA_RESOURCE_EXISTS: $(SERVICE_UPDATEGRAFANA_RESOURCE_EXISTS)
+      AZURE_RESOURCE_GROUP: $(AZURE_RESOURCE_GROUP)
+      GITHUB_PAT: $(GITHUB_PAT)
+
+  - task: AzureCLI@2
+    displayName: Deploy Application
+    inputs:
+      azureSubscription: DevOpsAzureRMConnection
+      scriptType: bash
+      scriptLocation: inlineScript
+      keepAzSessionActive: true
+      inlineScript: |
+        azd deploy --no-prompt
```

Comment:
not sure why, but when deploying at one of the customers I had to add `chmod +x ./scripts/*.sh`

they were deploying on Ubuntu and getting 126 error

do we need to add that to azure.yml?

Reply:
hmmm, I just pulled this branch and looked at the permissions, the execution bit is already set

![image](https://github.com/user-attachments/assets/1f1b08a2-112a-42f5-8d17-62aa1bacde40)

I also tried explicitly setting it and git doesn't detect a difference

Reply:
I know... I never had this issue. Maybe it was because how files were copied from our repo?
I'll mark this comment as resolved since maybe it was just an edge case?